Design/Mantid_As_a _SciPy_Styled_Package.md

```diff
@@ -0,0 +1,259 @@
+1. APIs will need to have some mechanism to ensure they match that specific library, for example by comparing the modified date time of the library.
+1. The APIs will need to be updated or removed when installing a different (new or older) version of Mantid, this should also include deleting APIs of Libraries that have been removed.
+
+
+#### Python Algorithms ####
+
+The registration of Python Algorithms and creation of the run time API is currently planned to be left as is, purely done at run time.
+
+Documentation Structure
+-----------------------
+
+### Top Level ###
+The documentation of Mantid will be rearranged to follow this structure:
+
+* Concepts
+* Simple scripting in Mantid (`mantid.simpleapi`)
+* Mathematical operations (`mantid.math`)
+* Instrument manipulations (`mantid.instrument`)
+* Workspace manipulations (`mantid.workspace`)
+* Experimental metadata (`mantid.metadata`)
+* Optimization and Curve Fitting (`mantid.optimization`)
+* File IO (`mantid.io`)
+* Common scientific Constants (`mantid.constants`)
+* Common neutron operations (`mantid.neutrons`)
+* Common Muon operations (`mantid.muons`)
+* Working with simulation codes (`mantid.simulations`)
+* Facility Specific Libaries (`mantid.sns` etc)
+* User Interfaces
+* Release Notes
+* API Reference
+    - Python 
+    - C++ <http://doxygen.mantidproject.org/>
+    - Full list of Algorithms
+    - Full list of Fit Functions
+    - Full list of Fit Minimizers
+
+### Module Page ###
+
+Each module page would not be auto generated as now, but would have sections within it that are auto generated within sphinx as we currently do within the algorithm pages, this wold be done by new custom directives.
+
+An example module page would contain the following:
+* The Title
+* A short description of the module and its purpose
+* A list of sub modules
+* A list of Algorithms within this module
+* A list of Fit functions within this module
+* Other objects
+*  An example or two of things that are possible using this module with python examples that are doc tested
+
+Each of the lists will be generated by a directive, and should not appear at all if there is nothing in the list.
+?? is it possible to extract the short module description from a docstring? ?? - If so all of the top can be replaced with a single directive.
+
+### Algorithm and Fit Function Pages ###
+
+These will remain as is, except that the category links will be redirected to the module pages.
+
+C++ Library Structure
+---------------------
+
+In order to make the C++ library easy to navigate we will rearrange the library to match the module structure of the python modules.  This will increase the number of libraries we have, but will help split out some of those that have become bloated, such as Algorithms and DataHandling.
+
+?? Do we want separate C++ libaries for sub-modules such as mantid.math.axes or should be stop at a certain level and gather all the mantid.math into a single library? ??
```

Comment:
As mentioned in a previous comment I think smaller libraries will allow us more flexibility and for instance would mean that we don't have to link many more dependencies than are required by that module.